--- cc/private/toolchain/windows_cc_toolchain_config.bzl
+++ cc/private/toolchain/windows_cc_toolchain_config.bzl
@@ -89,7 +89,7 @@ all_link_actions = [
 ]
 
 def _use_msvc_toolchain(ctx):
-    return ctx.attr.cpu in ["x64_windows", "arm64_windows"] and (ctx.attr.compiler == "msvc-cl" or ctx.attr.compiler == "clang-cl")
+    return ctx.attr.cpu in ["x64_windows", "x64_x86_windows", "arm64_windows"] and (ctx.attr.compiler == "msvc-cl" or ctx.attr.compiler == "clang-cl")
 
 def _impl(ctx):
     if _use_msvc_toolchain(ctx):
@@ -767,7 +767,6 @@
                         flag_group(
                             flags = [
                                 "/DNOMINMAX",
-                                "/D_WIN32_WINNT=0x0601",
                                 "/D_CRT_SECURE_NO_DEPRECATE",
                                 "/D_CRT_SECURE_NO_WARNINGS",
                                 "/bigobj",
@@ -777,7 +776,7 @@
                                 "/wd4291",
                                 "/wd4250",
                                 "/wd4996",
-                            ],
+                            ] + ctx.attr.default_compile_flags,
                         ),
                     ],
                 ),
