--- rules.bzl
+++ rules.bzl
@@ -25,8 +25,10 @@
     """
     ndk_path = ctx.attr.path or ctx.os.environ.get("ANDROID_NDK_HOME", None)
     if not ndk_path:
-        fail("Either the ANDROID_NDK_HOME environment variable or the " +
-             "path attribute of android_ndk_repository must be set.")
+        print("Either the ANDROID_NDK_HOME environment variable or the " +
+              "path attribute of android_ndk_repository must be set.")
+        ctx.file("BUILD.bazel", "")
+        return
     if ndk_path.startswith("$WORKSPACE_ROOT"):
         ndk_path = str(ctx.workspace_root) + ndk_path.removeprefix("$WORKSPACE_ROOT")
 
