diff --git a/pylib/gyp/__init__.py b/pylib/gyp/__init__.py
--- a/pylib/gyp/__init__.py
+++ b/pylib/gyp/__init__.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # Copyright (c) 2012 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
diff --git a/pylib/gyp/mac_tool.py b/pylib/gyp/mac_tool.py
--- a/pylib/gyp/mac_tool.py
+++ b/pylib/gyp/mac_tool.py
@@ -136,13 +136,13 @@ class MacTool(object):
     #     CFPropertyListCreateFromXMLData(): Old-style plist parser: missing
     #     semicolon in dictionary.
     # on invalid files. Do the same kind of validation.
-    import CoreFoundation
+    # import CoreFoundation
     with open(source, 'rb') as f:
       s = f.read()
-    d = CoreFoundation.CFDataCreate(None, s, len(s))
-    _, error = CoreFoundation.CFPropertyListCreateFromXMLData(None, d, 0, None)
-    if error:
-      return
+    # d = CoreFoundation.CFDataCreate(None, s, len(s))
+    # _, error = CoreFoundation.CFPropertyListCreateFromXMLData(None, d, 0, None)
+    # if error:
+    #   return

     with open(dest, 'wb') as fp:
       fp.write(s.decode(input_code).encode('UTF-16'))
