--- third_party/protobuf_legacy_opensource/src/google/protobuf/stubs/common.h
+++ third_party/protobuf_legacy_opensource/src/google/protobuf/stubs/common.h
@@ -122,7 +122,7 @@
 #define GOOGLE_PROTOBUF_VERIFY_VERSION                                    \
   ::google::protobuf_opensource::internal::VerifyVersion(                            \
     GOOGLE_PROTOBUF_VERSION, GOOGLE_PROTOBUF_MIN_LIBRARY_VERSION,         \
-    __FILE__)
+    "_FILE_")
 
 
 // ===================================================================
--- third_party/protobuf_legacy_opensource/src/google/protobuf/stubs/common_unittest.cc
+++ third_party/protobuf_legacy_opensource/src/google/protobuf/stubs/common_unittest.cc
@@ -95,9 +95,9 @@
 
   string text = GetCapturedTestStderr();
   EXPECT_EQ(
-    "[libprotobuf INFO " __FILE__ ":" + SimpleItoa(line + 1) + "] A message.\n"
-    "[libprotobuf WARNING " __FILE__ ":" + SimpleItoa(line + 2) + "] A warning.\n"
-    "[libprotobuf ERROR " __FILE__ ":" + SimpleItoa(line + 3) + "] An error.\n",
+    "[libprotobuf INFO " "_FILE_" ":" + SimpleItoa(line + 1) + "] A message.\n"
+    "[libprotobuf WARNING " "_FILE_" ":" + SimpleItoa(line + 2) + "] A warning.\n"
+    "[libprotobuf ERROR " "_FILE_" ":" + SimpleItoa(line + 3) + "] An error.\n",
     text);
 }
 
@@ -128,10 +128,10 @@
 
   ASSERT_EQ(2, captured_messages_.size());
   EXPECT_EQ(
-    "2 " __FILE__ ":" + SimpleItoa(start_line + 1) + ": An error.",
+    "2 " "_FILE_" ":" + SimpleItoa(start_line + 1) + ": An error.",
     captured_messages_[0]);
   EXPECT_EQ(
-    "1 " __FILE__ ":" + SimpleItoa(start_line + 2) + ": A warning.",
+    "1 " "_FILE_" ":" + SimpleItoa(start_line + 2) + ": A warning.",
     captured_messages_[1]);
 }
 
@@ -154,10 +154,10 @@
 
   ASSERT_EQ(2, captured_messages_.size());
   EXPECT_EQ(
-    "0 " __FILE__ ":" + SimpleItoa(line1) + ": Visible1",
+    "0 " "_FILE_" ":" + SimpleItoa(line1) + ": Visible1",
     captured_messages_[0]);
   EXPECT_EQ(
-    "0 " __FILE__ ":" + SimpleItoa(line2) + ": Visible2",
+    "0 " "_FILE_" ":" + SimpleItoa(line2) + ": Visible2",
     captured_messages_[1]);
 }
 
--- third_party/protobuf_legacy_opensource/src/google/protobuf/stubs/logging.h
+++ third_party/protobuf_legacy_opensource/src/google/protobuf/stubs/logging.h
@@ -146,7 +146,7 @@
 #define GOOGLE_LOG(LEVEL)                          \
   ::google::protobuf_opensource::internal::LogFinisher() = \
       ::google::protobuf_opensource::internal::LogMessage( \
-          ::google::protobuf_opensource::LOGLEVEL_##LEVEL, __FILE__, __LINE__)
+          ::google::protobuf_opensource::LOGLEVEL_##LEVEL, "_FILE_", __LINE__)
 #define GOOGLE_LOG_IF(LEVEL, CONDITION) \
   !(CONDITION) ? (void)0 : GOOGLE_LOG(LEVEL)
 
@@ -172,7 +172,7 @@
 }  // namespace internal
 #define GOOGLE_CHECK_NOTNULL(A)               \
   ::google::protobuf_opensource::internal::CheckNotNull( \
-      __FILE__, __LINE__, "'" #A "' must not be nullptr", (A))
+      "_FILE_", __LINE__, "'" #A "' must not be nullptr", (A))
 
 #ifdef NDEBUG
 
